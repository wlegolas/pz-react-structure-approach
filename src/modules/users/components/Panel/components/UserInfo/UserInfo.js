import React from 'react'

const UserInfo = () => {
  return (
    <div className="pz-user-container">
      <div className="pz-user-info">
        <div className="pz-box pz-left-box">
          <img
            src="https://avatars2.githubusercontent.com/u/698363?v=4"
            alt="GiHhub avatar"
          />
        </div>
        <div className="pz-box pz-right-box">
          <h1>Weslley Alves de Oliveira</h1>
          <p className="pz-label">Following: 1</p>
          <p className="pz-label">Followers: 2</p>
        </div>
      </div>
    </div>
  )
}

export default UserInfo
