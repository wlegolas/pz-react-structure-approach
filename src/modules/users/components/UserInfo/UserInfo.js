import PropTypes from 'prop-types'
import React from 'react'

const UserInfo = ({ userInfo }) => {
  return (
    <div className="pz-user-container">
      <div className="pz-user-info">
        <div className="pz-box pz-left-box">
          <img src={userInfo.avatarUrl} alt="GiHhub avatar" />
        </div>
        <div className="pz-box pz-right-box">
          <h1>{userInfo.name}</h1>
          <p className="pz-label">Following: {userInfo.totalFollowing}</p>
          <p className="pz-label">Followers: {userInfo.totalFollowers}</p>
        </div>
      </div>
    </div>
  )
}

UserInfo.propTypes = {
  userInfo: PropTypes.shape({
    avatarUrl: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    totalFollowing: PropTypes.number.isRequired,
    totalFollowers: PropTypes.number.isRequired
  }).isRequired
}

export default UserInfo
