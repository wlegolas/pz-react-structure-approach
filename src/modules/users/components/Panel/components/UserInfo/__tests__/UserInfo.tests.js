import toJson from 'enzyme-to-json'
import React from 'react'
import UserInfo from '../UserInfo'

describe('<UserInfo />', () => {
  let wrapper

  beforeAll(() => {
    wrapper = mount(<UserInfo />)
  })

  describe('render()', () => {
    it('should render the component', () => {
      expect(toJson(wrapper)).to.matchSnapshot()
    })

    describe('verify the structure', () => {
      it('should have container section', () => {
        expect(wrapper.find('.pz-user-container')).to.have.lengthOf(1)
      })

      it('should have an image element with the user avatar', () => {
        expect(wrapper.find('img')).to.have.lengthOf(1)
      })

      it('should have the element with the user name', () => {
        expect(wrapper.find('.pz-user-container h1')).to.have.lengthOf(1)
      })
    })
  })
})
