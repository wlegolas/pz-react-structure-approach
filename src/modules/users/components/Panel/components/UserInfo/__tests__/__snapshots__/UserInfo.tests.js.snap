// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> render() should render the component 1`] = `
<UserInfo>
  <div
    className="pz-user-container"
  >
    <div
      className="pz-user-info"
    >
      <div
        className="pz-box pz-left-box"
      >
        <img
          alt="GiHhub avatar"
          src="https://avatars2.githubusercontent.com/u/698363?v=4"
        />
      </div>
      <div
        className="pz-box pz-right-box"
      >
        <h1>
          Weslley Alves de Oliveira
        </h1>
        <p
          className="pz-label"
        >
          Following: 1
        </p>
        <p
          className="pz-label"
        >
          Followers: 2
        </p>
      </div>
    </div>
  </div>
</UserInfo>
`;
